test/sh/nway.sh

```diff
@@ -270,32 +290,123 @@ get_pods_on_nodes() {
+        if [ $? -ne 0 ]; then
+            echo "Failed to set permissions or list directory"
+            return 1
+        fi
+    fi
+
+    echo "Scripts successfully copied to $address:$remoteScriptDir"
+    return 0
+}
+
 failovercount=0
 # Fails a node give a list of nodes
 failnodes() {
-	nodelist=$*
-	# Fail all the nodes
-	for node in $nodelist
-	do
-		# kill node
-		if [ $BOUNCEKUBELETSECONDS -gt 0 ]; then
-			echo bouncing kubelet $node $COUNT
-			ssh $node nohup sh /root/bounce.kubelet --seconds $BOUNCEKUBELETSECONDS &
-		else
-			echo bouncing ip $node $COUNT
-			ssh $node nohup sh /root/bounce.ip --seconds $BOUNCEIPSECONDS &
-		fi
-	done
-	failovercount=$(expr $failovercount + 1)
-
-	if [ $REBOOT != "on" ]; then return; fi
-
-	sleep 60
-	for node in $nodelist
-	do
-		echo rebooting node $node
-		ssh $node nohup sh reboot.node
-	done
+    nodelist=$*
+    # Fail all the nodes
+    for node in $nodelist
+    do
+        copyOverTestScriptsToNode "$node"
+
+        # Kill node
+        if [ $BOUNCEKUBELETSECONDS -gt 0 ]; then
+            echo bouncing kubelet $node $COUNT
+            if [ "$NODE_USER" == "core" ]; then
+                ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /usr/tmp/karavi-resiliency-tests/bounce.kubelet --seconds $BOUNCEKUBELETSECONDS > /dev/null 2>&1 &"
+            elif [ "$NODE_USER" == "root" ]; then
+                sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /root/karavi-resiliency-tests/bounce.kubelet --seconds $BOUNCEKUBELETSECONDS > /dev/null 2>&1 &"
+            else
+                echo "Unsupported NODE_USER: $NODE_USER"
+                return 1
+            fi
+        else
+            echo bouncing ip $node $COUNT
+            if [ "$NODE_USER" == "core" ]; then
+                ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /usr/tmp/karavi-resiliency-tests/bounce.ip --seconds $BOUNCEIPSECONDS > /dev/null 2>&1 &"
+            elif [ "$NODE_USER" == "root" ]; then
+                sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /root/karavi-resiliency-tests/bounce.ip --seconds $BOUNCEIPSECONDS > /dev/null 2>&1 &"
+            else
+                echo "Unsupported NODE_USER: $NODE_USER"
+                return 1
+            fi
+        fi
+    done
+    failovercount=$(expr $failovercount + 1)
+
+    if [ $REBOOT != "on" ]; then return; fi
+
+    sleep 60
+    for node in $nodelist
+    do
+        echo rebooting node $node
+        if [ "$NODE_USER" == "core" ]; then
+            ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /usr/tmp/karavi-resiliency-tests/reboot.node > /dev/null 2>&1 &"
+        elif [ "$NODE_USER" == "root" ]; then
+            sshpass -p "$PASSWORD" ssh -o StrictHostKeyChecking=no "$NODE_USER@$node" "nohup sudo sh /root/karavi-resiliency-tests/reboot.node > /dev/null 2>&1 &"
```

Comment:
`ssh -o StrictHostKeyChecking=no` is being referenced a lot. Could you store this in a variable? Essentially, anything that is referenced more than once can be stored and reused for easier maintanence